diff --git gptfdisk-clean/diskio-unix.cc gptfdisk-workdir/diskio-unix.cc
index 0897c56..db76784 100644
--- gptfdisk-clean/diskio-unix.cc
+++ gptfdisk-workdir/diskio-unix.cc
@@ -37,7 +37,7 @@
 
 using namespace std;
 
-#if defined(__APPLE__) || defined(__linux__)
+#if defined(__APPLE__) || defined(__linux__) || defined(__zinnia__)
 #define off64_t off_t
 #define stat64 stat
 #define fstat64 fstat
@@ -177,7 +177,7 @@ int DiskIO::GetBlockSize(void) {
 #if defined (__FreeBSD__) || defined (__FreeBSD_kernel__)
       err = ioctl(fd, DIOCGSECTORSIZE, &blockSize);
 #endif
-#ifdef __linux__
+#if defined(__linux__) || defined(__zinnia__)
       err = ioctl(fd, BLKSSZGET, &blockSize);
 #endif
 
@@ -290,7 +290,7 @@ int DiskIO::DiskSync(void) {
            << "You should reboot or remove the drive.\n";
       platformFound++;
 #endif
-#ifdef __linux__
+#if defined(__linux__) || defined(__zinnia__)
       sleep(1); // Theoretically unnecessary, but ioctl() fails sometimes if omitted....
       fsync(fd);
       i = ioctl(fd, BLKRRPART);
@@ -466,6 +466,13 @@ uint64_t DiskIO::DiskSize(int *err) {
       sectors = bytes / b;
       platformFound++;
 #endif
+#ifdef __zinnia__
+      unsigned long long zb = 0;
+      *err = ioctl(fd, BLKGETSIZE64, &zb);
+      if (*err == 0)
+         sectors = zb / (unsigned long long) GetBlockSize();
+      platformFound++;
+#endif
 #ifdef __linux__
       long sz;
       long long b;
diff --git gptfdisk-clean/guid.cc gptfdisk-workdir/guid.cc
index d3e4fd5..0a41679 100644
--- gptfdisk-clean/guid.cc
+++ gptfdisk-workdir/guid.cc
@@ -16,6 +16,7 @@
 
 #include <stdio.h>
 #include <time.h>
+#include <unistd.h>
 #include <string.h>
 #include <string>
 #include <iostream>
@@ -141,6 +142,13 @@ void GUIDData::Zero(void) {
 void GUIDData::Randomize(void) {
    int i, uuidGenerated = 0;
 
+#if defined (__zinnia__)
+   if (getentropy(uuidData, sizeof(uuidData)) == 0) {
+      uuidData[7] = (uuidData[7] & 0x0F) | 0x40;
+      uuidData[8] = (uuidData[8] & 0x3F) | 0x80;
+      uuidGenerated = 1;
+   } // if
+#endif
 #if defined (_UUID_UUID_H) || defined (_UL_LIBUUID_UUID_H)
    uuid_generate(uuidData);
    ReverseBytes(&uuidData[0], 4);
diff --git gptfdisk-clean/guid.h gptfdisk-workdir/guid.h
index 515a75d..75cc25d 100644
--- gptfdisk-clean/guid.h
+++ gptfdisk-workdir/guid.h
@@ -25,6 +25,8 @@
    #pragma comment(lib, "Rpcrt4.lib")
    #endif
 typedef unsigned char my_uuid_t[16];
+#elif defined(__zinnia__)
+typedef unsigned char my_uuid_t[16];
 #else // Not Windows
 #include <uuid/uuid.h>
 typedef uuid_t my_uuid_t;
diff --git gptfdisk-clean/gptcurses.cc gptfdisk-workdir/gptcurses.cc
index b476700..35d64b2 100644
--- gptfdisk-clean/gptcurses.cc
+++ gptfdisk-workdir/gptcurses.cc
@@ -23,7 +23,7 @@
 #include <iostream>
 #include <string>
 #include <sstream>
-#if defined (__APPLE__) || (__FreeBSD__)
+#if defined (__APPLE__) || defined (__FreeBSD__) || defined (__zinnia__)
 #include <ncurses.h>
 #else
 #include <ncursesw/ncurses.h>
