diff --git dolphin-clean/src/kitemviews/private/kdirectorycontentscounterworker.cpp dolphin-workdir/src/kitemviews/private/kdirectorycontentscounterworker.cpp
index 8a69fb4..c8ec1d2 100644
--- dolphin-clean/src/kitemviews/private/kdirectorycontentscounterworker.cpp
+++ dolphin-workdir/src/kitemviews/private/kdirectorycontentscounterworker.cpp
@@ -8,7 +8,7 @@
 #include "kdirectorycontentscounterworker.h"
 
 // Required includes for countDirectoryContents():
-#if defined(Q_OS_WIN) || defined(Q_OS_HAIKU)
+#if defined(Q_OS_WIN) || defined(Q_OS_HAIKU) || defined(Q_OS_ZINNIA)
 #include <QDir>
 #else
 #include <QElapsedTimer>
@@ -23,7 +23,7 @@ KDirectoryContentsCounterWorker::KDirectoryContentsCounterWorker(QObject *parent
     qRegisterMetaType<KDirectoryContentsCounterWorker::Options>();
 }
 
-#if !defined(Q_OS_WIN) && !defined(Q_OS_HAIKU)
+#if !defined(Q_OS_WIN) && !defined(Q_OS_HAIKU) && !defined(Q_OS_ZINNIA)
 void KDirectoryContentsCounterWorker::walkDir(const QString &dirPath, bool countHiddenFiles, uint allowedRecursiveLevel)
 {
     QByteArray text = dirPath.toLocal8Bit();
@@ -138,7 +138,7 @@ void KDirectoryContentsCounterWorker::countDirectoryContents(const QString &path
 {
     const bool countHiddenFiles = options & CountHiddenFiles;
 
-#if defined(Q_OS_WIN) || defined(Q_OS_HAIKU)
+#if defined(Q_OS_WIN) || defined(Q_OS_HAIKU) || defined(Q_OS_ZINNIA)
     QDir dir(path);
     QDir::Filters filters = QDir::NoDotAndDotDot | QDir::System | QDir::AllEntries;
     if (countHiddenFiles) {
