diff --git kirigami-clean/CMakeLists.txt kirigami-workdir/CMakeLists.txt
index 36b9a2c..b7d82cd 100644
--- kirigami-clean/CMakeLists.txt
+++ kirigami-workdir/CMakeLists.txt
@@ -182,6 +182,6 @@ ecm_install_po_files_as_qm(poqm)
 include(ECMFeatureSummary)
 ecm_feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
-configure_file(qmllint.ini.in ${CMAKE_SOURCE_DIR}/.qmllint.ini)
+configure_file(qmllint.ini.in ${CMAKE_BINARY_DIR}/.qmllint.ini)
 
 kde_configure_git_pre_commit_hook(CHECKS CLANG_FORMAT)
