diff --git rust-libloading-09-clean/src/os/unix/consts.rs rust-libloading-09-workdir/src/os/unix/consts.rs
index 46fd1a1..7899ebc 100644
--- rust-libloading-09-clean/src/os/unix/consts.rs
+++ rust-libloading-09-workdir/src/os/unix/consts.rs
@@ -88,6 +88,7 @@ mod posix {
             target_os = "nto",
             target_os = "hurd",
             target_os = "cygwin",
+            target_os = "zinnia",
         ))] {
             pub(super) const RTLD_LAZY: c_int = 1;
         } else {
@@ -128,6 +129,7 @@ mod posix {
             target_os = "nto",
             target_os = "hurd",
             target_os = "cygwin",
+            target_os = "zinnia",
         ))] {
             pub(super) const RTLD_NOW: c_int = 2;
         } else if #[cfg(all(target_os = "android",target_pointer_width = "32"))] {
@@ -152,6 +154,7 @@ mod posix {
             all(target_os = "linux", target_arch = "mips"),
             all(target_os = "linux", target_arch = "mips64"),
             target_os = "cygwin",
+            target_os = "zinnia",
         ))] {
             pub(super) const RTLD_GLOBAL: c_int = 4;
         } else if #[cfg(any(
@@ -227,6 +230,7 @@ mod posix {
             target_os = "redox",
             target_os = "hurd",
             target_os = "cygwin",
+            target_os = "zinnia",
         ))] {
             pub(super) const RTLD_LOCAL: c_int = 0;
         } else {
